### PR TITLE
CLI: Parse pnp paths in storybook metadata

### PR DESCRIPTION
**Pull request by yannbf**
Closes #

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/lib/telemetry/src/storybook-metadata.test.ts [hunk 1]
```diff
@@ -82,26 +82,85 @@ describe('sanitizeAddonName', () => {
     // @ts-expect-error the property is read only but we can change it for testing purposes
     path.sep = '\\';
     const cwdMockPath = `C:\\Users\\username\\storybook-app`;
-    jest.spyOn(process, `cwd`).mockImplementationOnce(() => cwdMockPath);
+    const cwdSpy = jest.spyOn(process, `cwd`).mockReturnValue(cwdMockPath);
 
     expect(sanitizeAddonName(`${cwdMockPath}\\local-addon\\themes.js`)).toEqual(
       '$SNIP\\local-addon\\themes'
     );
+    cwdSpy.mockRestore();
```
Comment (shilman) [inline code/lib/telemetry/src/storybook-metadata.test.ts:90]: 1. Do we need this if `cwdSpy` is a local variable?
2. If so, is it guaranteed that this line will execute given that there could be an error on one of the previous lines?

File: code/lib/telemetry/src/get-framework-info.ts [hunk 2]
```diff
@@ -30,20 +33,39 @@ function findMatchingPackage(packageJson: PackageJson, suffixes: string[]) {
...
-  const { framework: frameworkInput } = mainConfig;
+export const getFrameworkPackageName = (mainConfig?: StorybookConfig) => {
+  const packageNameOrPath =
+    typeof mainConfig?.framework === 'string' ? mainConfig.framework : mainConfig?.framework?.name;
+
+  if (!packageNameOrPath) {
+    return null;
+  }
+
+  const normalizedPath = path.normalize(packageNameOrPath).replace(new RegExp(/\\/, 'g'), '/');
```
Comment (shilman) [inline code/lib/telemetry/src/get-framework-info.ts:44]: Do we need any Windows tests?

**Review by shilman — COMMENTED** [review]
Great catch @yannbf! Minor comments below

File: code/lib/telemetry/src/get-framework-info.ts (hunk 2 above)
Comment (yannbf) [reply code/lib/telemetry/src/get-framework-info.ts:44]: Great catch. Added!

File: code/lib/telemetry/src/storybook-metadata.test.ts (hunk 1 above)
Comment (yannbf) [reply code/lib/telemetry/src/storybook-metadata.test.ts]: Good one, I changed the code in a way that it will always restore the mocks even if it errors

**Review by shilman — APPROVED** [review]
Great job! 🎉

**Review by koonkoon — APPROVED** [review]

**Pull request merged**